framework/PageCarousel/PageCarousel.tsx

```diff
@@ -19,23 +29,38 @@ const SlideContainer = styled.div`
   }
 `;
 
+const CarouselContext = createContext<{ width: number; visibleCards: number }>({
+  width: 0,
+  visibleCards: 0,
+});
+
+/** This hook provides real time context of the carousel's width and the number
+ * of visible cards. The information may be needed for adjusting the widths of cards
+ * that show up inside the carousel.
+ */
+export function useCarouselContext() {
```

Comment:
@nixocio This error would only be caught if we wrap the call to the hook in a try/catch which is not a pattern used with hooks. Since `PageCarousel` itself wraps its returned JSX in the `CarouselContext.Provider`, the only scenario where I think the Context.Provider could be missing is if the `useCarouselContext` hook is called without rendering a `PageCarousel` component. However, even in this case the hook will return the default value that was set when the context was created `{
  width: 0,
  visibleCards: 0,
} `, which seems reasonable. I tested this by commenting out `PageCarousel` in `CollectionCategories.tsx` and the context defaulted to the initial value. So I think this check would be unnecessary. Let me know what you think.

Reply:
@vidyanambiar thanks , this one is a suggestion for the developer itself, to avoid to attempt to use a hook without a provider, since the Provider is not wrapping the whole component tree, using this hook in a component that is not wrapped by the provider would just return the default values, like failing silently.  I noticed that we have both patterns in the code basis. Perhaps, we could pick one. Not a blocker.

Reply:
Based on my test above where I removed the context provider (PageCarousel) there was no silent failure because the provider is wrapped in the PageCarousel itself. If the PageCarousel itself is removed, then it doesn't matter what this hook returns. It is only relevant in the context of a PageCarousel. 
Regarding the suggestion to throw an error if the context is undefined, unless we call this hook in a try/catch block it will still fail silently as an uncaught exception. It's usually not advisable to call hooks in a try/catch because try/catch is also a type of conditional and hooks should be invoked at the top level. And since a default value exists the context can never be be undefined. IMO validating context should not be handled in the hook.
You mentioned you've seen both patterns in the codebase. Could you point me to them please so I can get a better idea?

Reply:
Based on our chat offline, we should be safe here without this check since the provider is wrapped by the PageCarousel and we're not expecting devs to wrap their components in this provider.